### ISSUE TITLE
IDB wiped after v0.6.0 introduced Crosswalk update

### ISSUE THREAD

**Issue by felixhammerl** [MobileChromeApps/mobile-chrome-apps#554]
Hi guys,

apparently, 0.6.0 brought an update to a newer version of the Crosswalk webview, which wiped the existing IndexedDB our app uses. Is there any way to avoid this going forward? It kind of sucks to wipe our user's app databases because we receive a new version of the webview.

Cheers
Felix

**Comment by agrieve** [comment]
Ouch! That does suck! :(

@huningxin - do you know if this is a known issue with Crosswalk?

**Comment by tanx** [comment]
This is especially bad because we store the user's private PGP key in IDB. This bugs basically leads to complete data loss users don't have the key on another device :(

**Comment by agrieve** [comment]
@fujunwei - can you help with this?

**Comment by fujunwei** [comment]
I'd glad to do this.
I am trying to reproduce this issue.

**Comment by fujunwei** [comment]
I use cordova-plugin-indexedDB plugin to create databases and add new record. it works finely.
It would be helpful if you could describe test steps or simplify your usage scenario to reproduce the issue. Thanks.

**Comment by tanx** [comment]
Well we shiped the previous version of our using cca v0.5.1 which had an older version of the crosswalk webview if I understand correctly. Then we shiped the current version of our app using cca v0.6.0 which updated crosswalk. The IDB was empty for users that upgraded.

Is that enough info? Thanks

**Comment by fujunwei** [comment]
Do you mean users installed the apk which generated by cca v0.5.1, data can be stored int the IDB, but after users update the apk which generated by cca v0.6.0. the IDB was wiped?

**Comment by felixhammerl** [comment]
yes, exactly. we had a binary built with v0.5.1 in the appstore which allowed people to use idb. i assume the crosswalk update coming with v0.6.0 overwrote the pre-existing idb.

**Comment by agrieve** [comment]
For reference - v0.5.1 used Crosswalk v8 (Chrome 37), and v0.6.0 uses Crosswalk  v11 (Chrome 40).
Neither used cordova-plugin-indexedDB, but instead just used the built-in IndexedDB.

**Comment by fujunwei** [comment]
It can be reproduced  after upgrading app from Crosswalk 8 to Crosswalk 11, but it works finely from Crosswalk 9 to Crosswalk 11 and the same version upgrade.

**Comment by fujunwei** [comment]
We had approach to fix it, and created a JIRA ticket to keep track of this. 
https://crosswalk-project.org/jira/browse/XWALK-4047

**Comment by tanx** [comment]
Great thx for looking into this

**Comment by fujunwei** [comment]
The IndexedDB wasn't wiped but the upgraded Crosswalk didn't used it. You can find it in `/data/data/org.example.testing/app_xwalkcore/IndexedDB`. You can copy it manually under profile's subfolder `/data/data/org.example.testing/app_xwalkcore/Default/IndexedDB`, the name of the profile is "Default" by default.

**Comment by fujunwei** [comment]
@agrieve  Because Crosswalk 11 is unmaintained, Could you please upgrade crosswalk in `cca`?
